parachain/pallets/basic-channel/src/inbound/mod.rs

```diff
@@ -118,7 +118,7 @@ pub mod pallet {
 			})?;
 
 			let message_id = MessageId::Basic { account: envelope.account, nonce: envelope.nonce };
-			T::MessageDispatch::dispatch(envelope.source, message_id, &envelope.payload);
+			T::MessageDispatch::dispatch(envelope.source, message_id, &envelope.payload, Some(message.dispatch_weight));
 
 			<LatestVerifiedBlockNumber<T>>::set(block_number);
```

Comment:
Should move this before message dispatch, to make implementing benchmarks easier.